pkg/controller/networkpolicy/validate.go

```diff
@@ -309,6 +316,30 @@ func (v *antreaPolicyValidator) validateRuleName(ingress, egress []secv1alpha1.R
 	return isUnique(ingress) && isUnique(egress)
 }
 
+func (a *antreaPolicyValidator) validateAppliedTo(ingress, egress []secv1alpha1.Rule, specAppliedTo []secv1alpha1.NetworkPolicyPeer) (string, bool) {
+	appliedToInSpec := specAppliedTo != nil
+	countAppliedToInRules := func(rules []secv1alpha1.Rule) int {
+		num := 0
+		for _, rule := range rules {
+			if rule.AppliedTo != nil {
+				num++
+			}
+		}
+		return num
+	}
+	numAppliedToInRules := countAppliedToInRules(ingress) + countAppliedToInRules(egress)
+	if appliedToInSpec && (numAppliedToInRules > 0) {
+		return "appliedTo should not be set in both spec and rules", false
```

Comment:
Sorry, I must have missed some discussions. But why appliedTo cannot be on both rules and NP?
@abhiraut

Reply:
i think we decided to keep the implementation simple and not support a hybrid AppliedTo  .. initially i thought we could inherit the appliedTo from the spec in to  the rule if no AppliedTo set on the rule.. i am not sure if the implementation becomes too complex to support a hybrid setup?

Reply:
Spoke offline, at this moment we will keep the current implementation and disallow the hybrid case